### PR TITLE
Extend Project with Viewers

### LINKED ISSUE #984: robot-kubeconfig.yaml file with view permissions only
**What would you like to be added**:
As of know the robot-kubeconfig.yaml provided via the Gardener UI is one that allows user 'read' and some 'write' permissions on the shoot clusters.
We require a robot-kubeconfig.yaml that is allowed only 'read' permissions on the shoots. 

**Why is this needed**:
We want to access our shoots threw Gardener to get the Health Measurements Gardener collects on our shoots. To do so we need to access the shoots via a kubectl command on the seed to get the shoots data.

We wish to build a Monitoring cluster one for all our shoots and in order to access this information, it is required that we consume the robot-kubeconfig.yaml from the cluster (as a secret) or from source control.
Both options entail a security issue since the robot-kubeconfig.yaml can yield more than just read permissions.

### PR DESCRIPTION
**Pull request by vpnachev**
**What this PR does / why we need it**:
The `Project` resource is extended with optional field `Viewers` which gives read-only access to the gardener API.

**Which issue(s) this PR fixes**:
Fixes #984 

**Special notes for your reviewer**:
At the moment `Members` can create `ServiceAccounts`, but they cannot create roles and rolebindings. As alternative solution I was thinking about to let members create their own `RoleBindings` but in that way they could escalate their privileges(rolebinding to the clusterrole `cluster-admin`), so I decided not to do it in this way.

/cc @marwinski  @ThormaehlenFred 

**Release note**:

```noteworthy user
The `Project` resource is extended with the optional field `viewers` which is a list of subjects with read-only access (except `Secret`s) to the Gardener API.
```

### CODE REVIEW

**Comment by rfranzke** [comment]
@vlerenc how to proceed here?

**Comment by vlerenc** [comment]
I am not overly eager to merge that. On the one hand, people ask for this feature, so I understand it and thank you for the PR. On the other hand, I have not thought that through and have not planned how a compelling user management should look like.

Is there some strong opinion or has someone thought that through @marwinski or @ThormaehlenFred? If not, well... then I am also fine to merge it.

**Comment by marwinski** [comment]
I would like to have the pull request in but also understand that we are not quite sure on how user management should look like. How about waiting a few weeks until the UAM discussions/concept take some shape?

**Comment by vlerenc** [comment]
Not sure whether this will yield any tangible results ant time soon?

**Comment by ShimiT** [comment]
We are waiting for this PR to be merged soon, as we need it for our operator cluster solution (China delivery)

**Comment by vpnachev** [comment]
If the proposal for `viewers` is an issue for the UAM, then it is true for the already available `members`, because the only difference is only in the roles associated with the subjects(users). So this mean that we would have to deprecate `members` in the future and not to merge this PR?

**Comment by vlerenc** [comment]
Come on, we needed to start somewhere/with members, not roles, and not invent a massive UAM from the beginning before we even delivered any value.

Anyway, as I said, thank you for your proactive contribution and I am fine to merge it, because people ask for it. I just want to check whether a big change is coming any time soon in that field, because that's all API, once shipped. I hope you accept that reasoning for me asking @ThormaehlenFred and @marwinski.

You will also realise, that starting this journey (anything beyond the mere membership) will result in even more feature requests and configurability down the road. I have been there, it will happen.

**Comment by vpnachev** [comment]
First of all I want to say that the `members` are doing what was expected and perfectly fit for a simple user management that is needed as we see.
Second, as I have already said, personally I am fine if this PR does not get in. I just wanted to note that if another future UAM solution would have conflicts with `viewers`, most probably it will have the same conflicts with `members` and we have to have it in mind.
Lastly, you are right, this PR is changing the API(beta version) and we will not be able simply to remove the `viewers` if it is required in the future. So not merging this PR means less trouble for API deprecation.

I am writing this here in order to gather all of the relevant feedback on one place, which I hope will ease the taking of the final decision independently if it is positive or negative for the PR.

**Review by adracus — APPROVED** [review]
/lgtm
@ThormaehlenFred @marwinski please give feedback, from coding side this PR is also fine to merge.

**Review by rfranzke — APPROVED** [review]
/lgtm

**Comment by ThormaehlenFred** [comment]
I like the idea to introduce project members, who do only have read permissions with this PR. This is required as well to give internal and external auditors access with limited permissions. And it fits to the general UAM requirement discussion which is mentioned above _Define (more) fine-grained roles / authorizations for Gardener users / administrators / developers_ **(e.g. view only)**  Thus from my perspective this LGTM. But before merging you should get as well an LGTM from @marwinski and @ccwienk .

**Comment by adracus** [comment]
Ping @marwinski @ccwienk

**Pull request merged**
